### DISCUSSION TITLE
Shadows not visible - Task 3

### DISCUSSION THREAD

**Discussion by Robatic2021** [FieldRobotEvent/Virtual_Field_Robot_Event#15] [General]
In reality shadows are one of the most critical factors influencing the result of processing images taken outdoors. Gazebo has the ability to simulate shadows in the virtual maize field, so that allows for 'realistic' imagery coming from the camera in Task 3. 
However, when following the advised setup, running Ubuntu on a virtual machine, this visualisation of shadows does not work. A quick google search seems to point towards the fact that NVIDIA GPU's are not usable by a virtual machine. This idea is supported by our observations that the shadows are visible on a laptop (with a NVIDIA GPU ) with Ubuntu as main operating system, but not visible on a PC (without NVIDIA GPU) with Ubuntu as main operating system. 

If it is true that shadows are only visible on a PC with a (NVIDIA ?) GPU, running Ubuntu as the main operating system (not in a VM), this might be very limiting for teams in their ability to work on realistic results for Task 3. Therefore, we would like to ask whether this problem is recognised by others, and maybe then the organisation can express their opinion with respect to using the shadows in the competition. 

We look forward to hearing from you.

**Comment by Darkproduct** [comment]
Yes, on my Laptop I don't have an Nvidia GPU and the Gazebo lighting doesn't exist at all. You can use more CPU power to surcumvent some of the missing features, like GPU ray for a more realistic lidar, with `export LIBGL_ALWAYS_SOFTWARE=1`, but it seems like the Gazebo Team has no interesst in supporting anything other then Nvidia.

**Reply by Robatic2021** [reply]
Thanks for your response! Using LIBGL_ALWAYS_SOFTWARE does indeed give visible shadows, but they look rather different than the shadows on a laptop with a Nvidia GPU. Below I attached two images, to show the difference between the very faint and very pronounced shadows. I think the darker shadows have a bigger impact on image processing than the faint shadows, so maybe this is something to take into consideration.

![Screenshot from 2021-05-05 09-25-49](https://user-images.githubusercontent.com/82391920/117108941-32881700-ad84-11eb-8043-e2f518e91832.png)
![zachteschaduwen](https://user-images.githubusercontent.com/82391920/117109023-4fbce580-ad84-11eb-8850-e3e1b7866139.png)

**Reply by Darkproduct** [reply]
Try out the new world generator and check the shadows on the ground. I heard that they look much better than those on the old ground plane.

**Reply by Robatic2021** [reply]
Yes, true, they look a lot better. On some systems we now seem to see double shadows, but that does not seem to be a problem, so we're fine with that.
